paddle/fluid/platform/profiler.cc

```diff
@@ -102,7 +104,10 @@ Event::Event(EventType type, std::string name, uint32_t thread_id,
              const DeviceContext* dev_ctx)
     : type_(type), name_(name), thread_id_(thread_id), has_cuda_(false) {
 #ifdef PADDLE_WITH_CUDA
-  has_cuda_ = dev_ctx ? platform::is_gpu_place(dev_ctx->GetPlace()) : false;
+  has_cuda_ = (dev_ctx && g_state != ProfilerState::kCPU)
+                  ? platform::is_gpu_place(dev_ctx->GetPlace())
+                  : false;
```

Comment:
`has_cuda_` is confusing, the name should be changed, i.e.  -> `record_cuda_time_`

Reply:
If we do not use cuda event, then has_cuda_ is unnecessary.